unstructured_ingest/v2/processes/connectors/gitlab.py

```diff
@@ -0,0 +1,308 @@
+    download_config: GitLabDownloaderConfig
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["gitlab"], extras="gitlab")
+    def _fetch_content(self, path: str):
+        """Fetches the content of a file from the GitLab repository.
+
+        This method retrieves a file from the repository for the specified path and branch.
+        If the file is not found, it logs an error and raises the corresponding exception.
+
+        Args:
+            path (str): The path to the file within the repository.
+
+        Returns:
+            content_file: The file content object retrieved from the GitLab API.
+
+        Raises:
+            GitlabHttpError: If the specified file does not exist.
+        """
+        try:
+            project = self.connection_config.get_project()
+            ref_branch = self.connection_config.git_branch or project.default_branch
+            logger.info(f"Fetching file from path: {path!r} of branch: {ref_branch!r}")
+            content_file = project.files.get(
+                path,
+                ref=ref_branch,
+            )
+        except Exception as e:
+            logger.error(f"Failed to download: {e}")
+            raise e
+
+        return content_file
+
+    def _fetch_and_write(self, path: str, download_path: str) -> None:
+        """Fetches a file from the GitLab repository and writes its content to the specified path.
+
+        Args:
+            path (str): The path to the file within the repository.
+            download_path (str): The local path where the file will be saved.
+
+        Raises:
+            ValueError: If the file content could not be retrieved.
+
+        Notes:
+            - Decodes the file content before writing it to disk.
+            - Creates necessary parent directories if they do not exist.
+        """
+        content_file = self._fetch_content(path)
+        if content_file is None:
+            raise ValueError(
+                f"Failed to retrieve file from repo "
+                f"'{self.connection_config.url}/{path}'. Check logs.",
+            )
+        contents = content_file.decode()
+        logger.info(f"Writing download file to path: {download_path!r}")
+        with open(download_path, "wb") as f:
+            f.write(contents)
+
+    def run(self, file_data: FileData, **kwargs: Any) -> DownloadResponse:
+        """Downloads a file from the repository and returns a `DownloadResponse`.
+
+        Args:
+            file_data (FileData): Metadata about the file to be downloaded.
+            **kwargs (Any): Additional arguments (if required).
+
+        Returns:
+            DownloadResponse: A response object containing the download details.
+        """
+        download_path = self.get_download_path(file_data=file_data)
+        download_path.parent.mkdir(parents=True, exist_ok=True)
+
+        path = file_data.source_identifiers.fullpath
+        self._fetch_and_write(path, download_path)
```

Comment:
The above changes are needed because the download_path is of `Path` type, not a string.

Reply:
Done